foundation-extensions/mobile-core/mobile-core-api-reference.md

```diff
@@ -1,1182 +1,1805 @@
+  // method you should instantiate the window, load the UI into it and then make the window
+  // visible.
+  //
+  // You have 17 seconds to return from this method, or iOS will terminate your application.
+  //
+  public override bool FinishedLaunching(UIApplication app, NSDictionary options)
+  {
+    global::Xamarin.Forms.Forms.Init();
+    LoadApplication(new App());
+
+    // set the wrapper type
+    ACPCore.SetWrapperType(ACPMobileWrapperType.Xamarin);
+
+    // set launch config
+    ACPCore.ConfigureWithAppID("your-app-id");
+
+    // register SDK extensions
+    ACPIdentity.RegisterExtension();
+    ACPLifecycle.RegisterExtension();
+    ACPSignal.RegisterExtension();
+
+    // start core
+    ACPCore.Start(null);
+  }
+```
+
+**Android**
+
+```csharp
+using Com.Adobe.Marketing.Mobile;
+
+[Activity(Label = "TestApp", Icon = "@mipmap/icon", Theme = "@style/MainTheme", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation)]
+public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
+{
+  protected override void OnCreate(Bundle savedInstanceState)
+  {
+    base.OnCreate(savedInstanceState);
+
+    global::Xamarin.Forms.Forms.Init(this, savedInstanceState);
+    LoadApplication(new App());
+
+    // set the wrapper type
+    ACPCore.SetWrapperType(WrapperType.Xamarin);
+
+    // register SDK extensions
+    ACPCore.Application = this.Application;
+    ACPIdentity.RegisterExtension();
+    ACPLifecycle.RegisterExtension();
+    ACPSignal.RegisterExtension();
+
+    // start core
+    ACPCore.Start(null);
+}
+```
+{% endtab %}
+{% endtabs %}
+
+## registerURLHandler 
+
+Mobile SDK allows you to add a callback function that is triggered before the [`open url`](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/mobile-core/rules-engine#consequence-types) action occurs. If the callback function returns **Yes**, the SDK does not complete the `open url` action. If the callback function returns **No**, the SDK completes the `open url` action.
+
+{% tabs %}
+{% tab title="iOS \(ACP 2.x\)" %}
+**Objective C**
+
+### registerURLHandler
+
+**Syntax**
+
+```objectivec
++ (void) registerURLHandler: (nonnull BOOL (^) (NSString* __nullable url)) callback;
+```
+
+**Example**
+
+```objectivec
+[ACPCore registerURLHandler:^BOOL(NSString * _Nullable url) {
+    ...
+}];
+```
+{% endtab %}
+{% endtabs %}
+
+## resetIdentities
+
+The `resetIdentities` method requests that each extension resets the identities it owns. Each extension responds to this request uniquely.
+
+{% hint style="warning" %}
+This API is only handled by the [Edge](mobile-core-api-reference.md) and [Edge Identity](https://github.com/Adobe-Marketing-Cloud/aep-sdks-documentation/tree/de62274f59c14b23acae8992352ec63fbb16411b/foundation-extensions/adobe-edge-identity/README.md) extensions. Calling `resetIdentities` does **not** clear identities in non-edge extensions, such as [Identity](identity/).
+{% endhint %}
```

Comment:
As more extensions started handling this event, let's remove this section and update the above description to:
```
The `resetIdentities` method requests that each extension resets the identities it owns and each extension responds to this request uniquely. For more details, check the `resetIdentities` API reference on each of the extensions you use.